dashboard/src/shared/schema.ts

```diff
@@ -28,29 +29,42 @@ export function retrieveBasicFormParams(
       // The param path is its parent path + the object key
       const itemPath = `${parentPath || ""}${propertyKey}`;
       const { type, form } = properties[propertyKey];
-      // If the property has the key "form", it's a basic parameter
-      if (form) {
-        // Use the default value either from the JSON schema or the default values
-        const value = getValue(defaultValues, itemPath, properties[propertyKey].default);
-        const param: IBasicFormParam = {
-          ...properties[propertyKey],
-          path: itemPath,
-          type,
-          value,
-          enum: properties[propertyKey].enum?.map(
-            (item: { toString: () => any }) => item?.toString() ?? "",
-          ),
-          children:
-            properties[propertyKey].type === "object"
-              ? retrieveBasicFormParams(defaultValues, properties[propertyKey], `${itemPath}/`)
-              : undefined,
-        };
-        params = params.concat(param);
+      // If it is a kubeapps-specific schema
+      if (isLegacySchema) {
+        // If the property has the key "form", it's a basic parameter
+        if (form) {
+          params = params.concat(
+            getParam(defaultValues, itemPath, properties, propertyKey, type, isLegacySchema),
+          );
+        } else {
+          // If the property is an object, iterate recursively
+          if (schema.properties?.[propertyKey].type === "object") {
+            params = params.concat(
+              retrieveBasicFormParams(
+                isLegacySchema,
+                defaultValues,
+                properties[propertyKey],
+                `${itemPath}/`,
+              ),
+            );
+          }
+        }
+        // Otherwise, it is a normal JSON schema with no kubeapps-specific logic
       } else {
         // If the property is an object, iterate recursively
-        if (schema.properties![propertyKey].type === "object") {
+        if (schema.properties?.[propertyKey].type === "object") {
+          params = params.concat(
+            retrieveBasicFormParams(
+              isLegacySchema,
+              defaultValues,
+              properties[propertyKey],
+              `${itemPath}/`,
+            ),
+          );
+          // Unless there is a form property with a false value, we will render every field in the schema
+        } else if (form !== false) {
```

Comment:
We're supposing way too much... an OpenAPI Schema won't have any `form` property. But it does support extensions, so it could eventually become `x-kubeapps-form: true` and it would be openapi-compliant.

Reply:
More details at https://github.com/bitnami-labs/readme-generator-for-helm/issues/15